rarible-subgraph/src/exchange-v-2.ts

```diff
@@ -0,0 +1,420 @@
+    // rightAsset is NFT
+    log.info("{} {} {} {} address id and data and hash leftasset is nft", [rightAsset.address.toHexString(), rightAsset.id.toString(), orderRight.makeAsset.assetType.data.toHexString(), transactionHash.toHexString()]);
+
+    let nft = new airstack.nft.NFT(
+      rightAsset.address,
+      rightAssetType,
+      rightAsset.id,
+      orderRight.makeAsset.value,
+    )
+
+    let orderLeftInput = new LibOrder(
+      orderLeft.maker,
+      convertAssetToLibAsset(orderLeft.makeAsset),
+      orderLeft.taker,
+      convertAssetToLibAsset(orderLeft.takeAsset),
+      orderLeft.salt,
+      orderLeft.start,
+      orderLeft.end,
+      orderLeft.dataType,
+      orderLeft.data,
+    );
+
+    let orderRightInput = new LibOrder(
+      orderRight.maker,
+      convertAssetToLibAsset(orderRight.makeAsset),
+      orderRight.taker,
+      convertAssetToLibAsset(orderRight.takeAsset),
+      orderRight.salt,
+      orderRight.start,
+      orderRight.end,
+      orderRight.dataType,
+      orderRight.data,
+    );
+
+    let paymentSide = new LibDealSide(
+      convertAssetToLibAsset(orderLeft.makeAsset),
+      getOriginFeeArray(orderLeft.dataType, orderLeft.data, transactionHash).payoutFeeArray,
+      getOriginFeeArray(orderLeft.dataType, orderLeft.data, transactionHash).originFeeArray,
+      zeroAddress,
+      orderLeft.maker,
+    );
+
+    let nftSide = new LibDealSide(
+      convertAssetToLibAsset(orderRight.makeAsset),
+      getOriginFeeArray(orderRight.dataType, orderRight.data, transactionHash).payoutFeeArray,
+      getOriginFeeArray(orderRight.dataType, orderRight.data, transactionHash).originFeeArray,
+      zeroAddress,
+      orderRight.maker,
+    );
+
+    let matchAndTransferResult = matchAndTransfer(paymentSide, nftSide, orderLeftInput, orderRightInput, call.from, dataSource.address(), transactionHash);
+    log.info("{} {} {} match and transfer result for rightasset transaction hash {}", [matchAndTransferResult.originFee.value.toString(), matchAndTransferResult.royalty.value.toString(), matchAndTransferResult.payment.toString(), transactionHash.toHexString()]);
+
+    let nftSales = new airstack.nft.Sale(
+      orderLeft.maker,  //to
+      orderRight.maker, //from
+      nft, //nft
+      matchAndTransferResult.payment, //payment amount
+      leftAsset.address, //payment token
+      matchAndTransferResult.originFee.value,  //protocol fees
+      matchAndTransferResult.originFee.address, //protocol beneficiary
+      matchAndTransferResult.royalty.value, //royalty fee
+      matchAndTransferResult.royalty.address, //royalty beneficiary
+    )
+
+    airstack.nft.trackNFTSaleTransactions(
+      transactionHash.toHexString(),
+      call.transaction.index,
+      [nftSales],
+      AirProtocolType.NFT_MARKET_PLACE,
+      AirProtocolActionType.SELL,
+      call.block.timestamp,
+      call.block.number,
+      call.block.hash.toHexString()
+    );
+  } else {
```

Comment:
Seems logic of `else` is duplicate of `if` logic. Only variables are switched. Extract them in a function.